### DISCUSSION TITLE
Checking validity of generated types?

### DISCUSSION THREAD

**Discussion by Lilja** [dotansimha/graphql-code-generator#5791] [Ideas] [answered]
I would really like to use this project to generate types and what not. I looked around in the project and think that there are potential trouble when the types you generate is not the latest ones. 

Imagine if you delete a property from a type, delete it from whatever data source you have and not updating the types. Typescript wouldn't complain and the build would work, as expected.
What I would like to see, is that if this project could also ingest the hash of the file as some sort of string. Then add some sort of `yarn graphql-codegen --check` that would retrieve the hash, grep or sort of import the hash, compare it and finally check for validity, error if it's not matching. 
This would make CI easier as it would force the developer to be able to resolve that in their codebase.

**Comment by tvvignesh** [comment, marked as answer]
@dotansimha @ardatan I guess adding some sort of checksum to the generated file would solve this right?

- Generate the checksum of the final schema retrieved using the config file for every project separately
- Generate the checksum of every operation separately
- Add the checksum to the generated TS file
- Comparewith the files when running `--check` as suggested

What I am still thinking of though is: "Is it really required? Considering the fact that you can always regenerate the TS files and overwrite the types again keeping things in sync"

**Reply by dotansimha** [reply]
Totally agree with you here...
